### PR TITLE
Don't reencode reports in the collector

### PR DESCRIPTION
**Pull request by tomwilkie**
Part of #1789

### CODE REVIEW

File: app/collector.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ type Reporter interface {
 // Adder is something that can accept reports. It's a convenient interface for
 // parts of the app, and several experimental components.
 type Adder interface {
-	Add(context.Context, report.Report) error
+	Add(context.Context, report.Report, []byte) error
```
Comment (2opremio) [inline app/collector.go]: Can you please document what's the content and role of the `[]byte` parameter?

File: app/multitenant/memcache_client.go [hunk 2]
```diff
@@ -209,9 +209,14 @@ func (c *MemcacheClient) FetchReports(keys []string) (map[string]report.Report,
 func (c *MemcacheClient) StoreReport(key string, report *report.Report) (int, error) {
```
Comment (2opremio) [inline app/multitenant/memcache_client.go]: Is there a reason to keep the StoreReport function? Why not inline everything in StoreReportBytes?

Comment (rade) [inline app/multitenant/memcache_client.go]: Alternatively, make `StoreReport` take an extra (possibly nil) `[]byte` argument. To match `collector.Add`.

File: app/router.go [hunk 3]
```diff
@@ -136,6 +142,12 @@ func RegisterReportPostHandler(a Adder, router *mux.Router) {
 			return
 		}
 
+		// a.Add(..., buf) assumes buf is gzip'd msgpack
+		if !isMsgpack {
```
Comment (2opremio) [inline app/router.go]: Why not simply reject uncompressed msgpack reports? I think it's safe to do since we haven't done it in the past.

Comment (tomwilkie) [inline app/router.go]: discussed offline, fixprobe.

**Comment by jml** [comment]
LGTM.

File: docker/Dockerfile [hunk 4]
```diff
@@ -2,7 +2,7 @@ FROM alpine:3.3
 MAINTAINER Weaveworks Inc <help@weave.works>
 LABEL works.weave.role=system
 WORKDIR /home/weave
-RUN echo "http://dl-3.alpinelinux.org/alpine/edge/testing" >>/etc/apk/repositories && \
+RUN echo "http://dl-cdn.alpinelinux.org/alpine/edge/community" >>/etc/apk/repositories && \
```
Comment (rade) [inline docker/Dockerfile]: We may want to backport this to `release-0.17`.

**Pull request merged**